### PR TITLE
updated pzp's `Dockerfile` base image from `perl` to `debian`

### PR DESCRIPTION
**Pull request by mwarin**
Changes:

* Changed base image from `perl:5.38` to `debian:bookworm`.
* Updated dependencies to the point where `docker compose build` succeeds & tests pass.
* Switched to installing as many perl modules as possible with `apt` instead of `cpanm`.
* Removed some perl modules that already come with perl and do not need to be installed (e.g. `Data::Dumper`, `Exporter`, `Sys::Hostname` and more).
* Removed `cpanfile`... this is something I think only makes sense when distributing packages on CPAN, which we are not doing with this repo.

Fixes issue #39.

I don't have any other ideas for how best to test this (beyond the existing tests) and will leave that to the imagination of the reviewer.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,45 +1,52 @@
...
+    libnet-ssleay-perl \
+    libssl-dev \
+    msmtp-mta \
+    perl \
+    pigz \
+    ruby-dev
+
+# Install perl modules
+RUN cpanm --notest \
+    DBD::MariaDB \
```
Comment (aelkiss) [inline Dockerfile:27]: Some of these could be installed from Debian packages, but I'm not too fussed about trying to get them all that way right now.

**Review by aelkiss — APPROVED** [review]
Looks good to me, assuming the image builds & tests pass.

File: Dockerfile (hunk 1 above)
Comment (mwarin) [reply Dockerfile:27]: How would I go about figuring that out?

Comment (mwarin) [reply Dockerfile:27]: https://packages.debian.org/bookworm/perl/ also seems viable.

Comment (aelkiss) [reply Dockerfile:29]: Honestly I'm not sure of the best way to get a comprehensive list. You could look at what we're installing from Debian vs. CPAN for babel here -- https://github.com/hathitrust/babel/blob/main/Dockerfile#L6-L59 -- that may give some pointers. The list in feed may also be useful to look at -- some are installed via apt and then anything that didn't get is installed by dependencies listed in Makefile.PL (see here https://github.com/hathitrust/feed/blob/main/Makefile.PL and https://github.com/hathitrust/feed/blob/main/Dockerfile#L23-L44).

Comment (mwarin) [reply Dockerfile]: I've switched all perl modules that I could find from being installed with `cpanm` to being installed with `apt`. Removed a couple of modules that already come with perl (`Data::Dumper`, `Exporter`, `Sys::Hostname`).

File: Dockerfile [hunk 2]
```diff
@@ -1,45 +1,47 @@
...
+    libdate-manip-perl \
+    libdb-dev \
+    libdbd-mariadb-perl \
+    libdevel-cover-perl \
+    libjson-xs-perl \
+    libmarc-file-mij-perl \
+    libmarc-perl \
+    libmarc-record-perl \
+    libmarc-xml-perl \
+    libmariadb-dev \
```
Comment (aelkiss) [inline Dockerfile:25]: We likely don't need this if we're installing libdbd-mariadb-perl from apt

File: Dockerfile [hunk 3]
```diff
@@ -1,45 +1,47 @@
...
+    libxml-libxml-perl \
+    libyaml-libyaml-perl \
+    msmtp-mta \
+    perl \
+    pigz \
+    ruby-dev
+
+# Install perl modules that we cannot get with apt-get
+RUN cpanm --notest \
+    DB_File \
```
Comment (aelkiss) [inline Dockerfile:39]: `DB_File` comes with `apt install perl` (in the `libperl5.36` package)

File: Dockerfile [hunk 4]
```diff
@@ -1,45 +1,47 @@
...
 
+# Install Debian packages
+RUN apt-get update && apt-get install -y \
+    bsd-mailx \
+    build-essential \
+    cpanminus \
+    git \
+    libcrypt-ssleay-perl \
+    libdate-manip-perl \
+    libdb-dev \
```
Comment (aelkiss) [inline Dockerfile:17]: Probably not needed since we can get `DB_File` from the base perl library

File: Dockerfile [hunk 5]
```diff
@@ -1,45 +1,47 @@
...
+    msmtp-mta \
+    perl \
+    pigz \
+    ruby-dev
+
+# Install perl modules that we cannot get with apt-get
+RUN cpanm --notest \
+    DB_File \
+    Devel::Cover::Report::Coveralls \
+    Test::More \
```
Comment (aelkiss) [inline Dockerfile:41]: `Test::More` is part of the `perl-modules-5.36` package which is installed by default when you `apt install perl`

File: Dockerfile [hunk 6]
```diff
@@ -1,45 +1,47 @@
...
+    libdbd-mariadb-perl \
+    libdevel-cover-perl \
+    libjson-xs-perl \
+    libmarc-file-mij-perl \
+    libmarc-perl \
+    libmarc-record-perl \
+    libmarc-xml-perl \
+    libmariadb-dev \
+    libnet-ssleay-perl \
+    libssl-dev \
```
Comment (aelkiss) [inline Dockerfile:27]: Likely not needed since we are installing `libnet-ssleay-perl` from apt

File: Dockerfile [hunk 7]
```diff
@@ -1,45 +1,47 @@
...
 ENV ROOTDIR /usr/src/app
 ENV HOME /usr/src/app
 
 COPY . $ROOTDIR
 WORKDIR $ROOTDIR
 
+# Install Debian packages
+RUN apt-get update && apt-get install -y \
+    bsd-mailx \
+    build-essential \
```
Comment (aelkiss) [inline Dockerfile:12]: I would be curious if we could get away without `build-essential` or if one of the remaining CPAN modules or ruby gems still requires `make` and `gcc` and friends

Comment (mwarin) [reply Dockerfile:12]: works fine without

File: Dockerfile (hunk 4 above)
Comment (mwarin) [reply Dockerfile:17]: yep, works fine without

File: Dockerfile (hunk 3 above)
Comment (mwarin) [reply Dockerfile:39]: yep, works fine without this

File: Dockerfile (hunk 2 above)
Comment (mwarin) [reply Dockerfile:25]: yep, works fine without this

File: Dockerfile (hunk 5 above)
Comment (mwarin) [reply Dockerfile]: yep, trimmed this and it still works

**Review by aelkiss — APPROVED** [review]
`libdevel-cover` ends up bringing in `build-essential` anyway. Eventually, we probably only need that for dev/testing and might be able to slim down our production images, but that day is *not today*. This all looks good to me at this point.

**Pull request merged**